operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeployment.java

```diff
@@ -429,7 +429,10 @@ private KeycloakDistConfigurator configureDist() {
 
     private List<EnvVar> getEnvVars() {
         // default config values
-        List<ValueOrSecret> serverConfig = Constants.DEFAULT_DIST_CONFIG.entrySet().stream()
+        List<ValueOrSecret> serverConfig = Constants.DEFAULT_DIST_CONFIG
+                .entrySet()
+                .stream()
+                .sorted((e1, e2) -> e1.getKey().compareTo(e2.getKey()))
```

Comment:
Thanks @andre-nascimento-rh, good catch. As per [documentation](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Map.html), indeed we can't rely on the ordering of immutable maps:
> The iteration order of mappings is unspecified and is subject to change.

I am think whether we should be really sorting the options at each reconciliation loop. Maybe a more optimal approach would be to use a List of `ValueOrSecret` directly in the `DEFAULT_DIST_CONFIG`. That way we'd also optimize the current code avoiding mapping the constant Map to List in each reconciliation loop. Another option would be using an ordered Map, e.g. `LinkedHashMap`. But that might be a bit overkill. WDYT?

Reply:
I went for the first suggestion. Changed.

Thanks.